corehq/apps/reports/analytics/esaccessors.py

```diff
@@ -193,3 +193,69 @@ def get_form_counts_by_user_xmlns(domain, startdate, enddate, user_ids=None,
                 counts[key] = xmlns_bucket.doc_count
 
     return counts
+
+
+def get_form_duration_stats_by_user(
+        domain,
+        app_id,
+        xmlns,
+        user_ids,
+        startdate,
+        enddate,
+        by_submission_time=True):
+    """Gets stats on the duration of a selected form grouped by users"""
+    date_filter_fn = submitted_filter if by_submission_time else completed_filter
+
+    query = (
+        FormES()
+        .domain(domain)
+        .app(app_id)
+        .user_id(user_ids)
+        .xmlns(xmlns)
+        .filter(date_filter_fn(gte=startdate, lt=enddate))
+        .aggregation(
+            TermsAggregation('user_id', 'form.meta.userID').aggregation(
+                ExtendedStatsAggregation(
+                    'duration_stats',
+                    'form.meta.timeStart',
+                    script="doc['form.meta.timeEnd'].value - doc['form.meta.timeStart'].value",
+                )
+            )
+        )
+        .size(0)
+    )
+    result = {}
+    buckets_dict = query.run().aggregations.user_id.buckets_dict
+    for user_id, bucket_dict in buckets_dict.iteritems():
+        result[user_id] = bucket_dict.result['duration_stats']
+    return result
+
+
+def get_form_duration_stats_for_users(
+        domain,
+        app_id,
+        xmlns,
+        user_ids,
+        startdate,
+        enddate,
+        by_submission_time=True):
+    """Gets the form duration stats for a group of users"""
+    date_filter_fn = submitted_filter if by_submission_time else completed_filter
+
+    query = (
+        FormES()
+        .domain(domain)
+        .app(app_id)
+        .user_id(user_ids)
+        .xmlns(xmlns)
+        .filter(date_filter_fn(gte=startdate, lt=enddate))
+        .aggregation(
+            ExtendedStatsAggregation(
+                'duration_stats',
+                'form.meta.timeStart',
+                script="doc['form.meta.timeEnd'].value - doc['form.meta.timeStart'].value",
+            )
+        )
+        .size(0)
+    )
+    return query.run().aggregations.duration_stats.result
```

Comment:
https://github.com/dimagi/commcare-hq/commit/c885bbabbea9c1f08947b05a520b235bf3e6225d tried to return the `ExtendedStatsResult` but got ugly when have to create a default for it on this line: https://github.com/dimagi/commcare-hq/pull/10329/files#diff-30c4e1ef0d3ceb995aab05246948b579R807. so i reverted to just returning the dictionary